Expensify/App
# check if reports data is up to date before update last read action id

### Details
Verification added for updating the list of unread messages only if the reports data is up to date

### Fixed Issues
$ https://github.com/Expensify/App/issues/5405

### Tests
1. Open a chat in the browser with the purpose to create cached data
2. Close the tab
3. Open the same chat on an other device
4. Send a message to this chat
5. Open the same chat again on the browser used at the point 1
6. Verify that the message sent by ourself in the other device does not appear as unread
7. Repeat the same steps with the other devices

### QA Steps
The same steps as above

### Tested On


### Screenshots

#### Web
<img width="790" alt="web" src="https://user-images.githubusercontent.com/26542066/139244750-88e5e237-3848-44cb-8504-1b70cd2ec7a9.png">

#### Mobile Web
<img width="452" alt="mobile web" src="https://user-images.githubusercontent.com/26542066/139244800-cb9a65d6-1bfd-423c-abb5-7d1caf655ee7.png">

#### Desktop
<img width="1212" alt="desktop" src="https://user-images.githubusercontent.com/26542066/139244847-07dfd7ec-d6b0-4064-a27d-a0ac11cb4b03.png">

#### iOS
<img width="470" alt="ios" src="https://user-images.githubusercontent.com/26542066/139244889-6fb07c21-40cf-41aa-88d7-4f3cf6262f9e.png">

#### Android
![android](https://user-images.githubusercontent.com/26542066/139244951-c37c723d-b1fe-448f-8074-3880d0e7eaa3.jpg)
